test/r_tree/r_tree_test.dart

```diff
@@ -197,30 +213,105 @@ main() {
             tree.insert(item);
           }
         }
+        assertTreeHeightValidity(tree);
 
         var items = tree.search(Rectangle(0, 0, 50, 50));
         expect(items.length, equals(2500));
 
         data.forEach((RTreeDatum item) {
           tree.remove(item);
         });
+        assertTreeHeightValidity(tree);
 
         items = tree.search(Rectangle(0, 0, 50, 50));
         expect(items.length, equals(0));
 
         //test inserting after removal to ensure new root leaf node functions correctly
         tree.insert(RTreeDatum<String>(Rectangle(0, 0, 1, 1), 'New Initial Item'));
+        assertTreeHeightValidity(tree);
 
         items = tree.search(Rectangle(0, 0, 50, 50));
 
         items.forEach((datum) {
           expect(datum.value, equals('New Initial Item'));
         });
       });
+
+      test('remove all items and then reload', () {
```

Comment:
there are no asserts in this test, seems like there ought to be?

Reply:
This test is implicitly asserting the `load` doesn't fail. If it does, we get a failure like:
```
00:00 +10 -1: RTree Remove remove all items and then reload [E]                                                                                                                                           
  type 'LeafNode<dynamic>' is not a subtype of type 'RTreeDatum<dynamic>' of 'value'
  dart:core                                     List.add
  package:r_tree/src/r_tree/r_tree.dart 109:19  RTree._insertTree
  package:r_tree/src/r_tree/r_tree.dart 97:7    RTree.load
  test/r_tree/r_tree_test.dart 254:14           main.<fn>.<fn>.<fn>

```
I will add some assertions to strengthen the test besides this error check.

Reply:
I figured, but also thought it wouldn't hurt to use the new assert utility you added

Reply:
Oh haha, yeah I'll include that. :+1: